main.go

```diff
@@ -0,0 +1,251 @@
+				Usage: "Create backups after this time interval in minutes",
+				Value: 5 * time.Minute,
+			},
+			cli.DurationFlag{
+				Name:  "retention",
+				Usage: "Retain backups within this time interval in hours",
+				Value: 24 * time.Hour,
+			},
+			cli.BoolFlag{
+				Name:   "debug",
+				Usage:  "Verbose logging information for debugging purposes",
+				EnvVar: "RANCHER_DEBUG",
+			},
+			cli.BoolFlag{
+				Name:  "once",
+				Usage: "Take backup only once",
+			},
+			cli.StringFlag{
+				Name:  "name",
+				Usage: "Backup name to take once",
+			},
+			cli.StringFlag{
+				Name:   "cacert",
+				Usage:  "Etcd CA client certificate path",
+				EnvVar: "ETCD_CACERT",
+			},
+			cli.StringFlag{
+				Name:   "cert",
+				Usage:  "Etcd client certificate path",
+				EnvVar: "ETCD_CERT",
+			},
+			cli.StringFlag{
+				Name:   "key",
+				Usage:  "Etcd client key path",
+				EnvVar: "ETCD_KEY",
+			},
+		},
+	}
+}
+
+func SetLoggingLevel(debug bool) {
+	if debug {
+		log.SetLevel(log.DebugLevel)
+	} else {
+		log.SetLevel(log.InfoLevel)
+	}
+}
+
+func RollingBackupAction(c *cli.Context) error {
+	SetLoggingLevel(c.Bool("debug"))
+
+	creationPeriod := c.Duration("creation")
+	retentionPeriod := c.Duration("retention")
+	etcdCert := c.String("cert")
+	etcdCACert := c.String("cacert")
+	etcdKey := c.String("key")
+	if len(etcdCert) == 0 || len(etcdCACert) == 0 || len(etcdKey) == 0 {
+		return fmt.Errorf("Failed to find etcd cert or key paths")
+	}
+	log.WithFields(log.Fields{
+		"creation":  creationPeriod,
+		"retention": retentionPeriod,
+	}).Info("Initializing Rolling Backups")
+
+	if c.Bool("once") {
+		backupName := c.String("name")
+		if len(backupName) == 0 {
+			backupName = fmt.Sprintf("%s_etcd", time.Now().Format(time.RFC3339))
+		}
+		return CreateBackup(backupName, etcdCACert, etcdCert, etcdKey)
+	}
+	backupTicker := time.NewTicker(creationPeriod)
+	for {
+		select {
+		case backupTime := <-backupTicker.C:
+			backupName := fmt.Sprintf("%s_etcd", backupTime.Format(time.RFC3339))
+			CreateBackup(backupName, etcdCACert, etcdCert, etcdKey)
+			DeleteBackups(backupTime, retentionPeriod)
+		}
+	}
+}
+
+func CreateBackup(backupName string, etcdCACert, etcdCert, etcdKey string) error {
+	failureInterval := 15 * time.Second
+	backupDir := fmt.Sprintf("%s/%s", backupBaseDir, backupName)
+	var err error
+	for retries := 0; retries <= backupRetries; retries++ {
+		if retries > 0 {
+			time.Sleep(failureInterval)
+		}
+
+		clusterHealthy := true
+		// check if the cluster is healthy
+		cmd := exec.Command("etcdctl",
+			"--endpoints=[127.0.0.1:2379]",
+			"--cacert="+etcdCACert,
+			"--cert="+etcdCert,
+			"--key="+etcdKey,
+			"endpoint", "health")
+		data, err := cmd.CombinedOutput()
+
+		if strings.Contains(string(data), "unhealthy") {
+			clusterHealthy = false
+			log.WithFields(log.Fields{
+				"error": err,
+				"data":  string(data),
+			}).Warn("Checking member health failed from etcd member")
+			continue
+		}
+		if !clusterHealthy {
```

Comment:
looks like we continue on line 154, and  clusterHealthy would never be false at this line.